crates/api_crud/src/site/read.rs

```diff
@@ -21,46 +21,68 @@ use lemmy_utils::{
   error::{LemmyError, LemmyErrorExt, LemmyErrorType},
   version,
 };
+use moka::future::Cache;
+use once_cell::sync::Lazy;
+use std::time::Duration;
 
 #[tracing::instrument(skip(context))]
 pub async fn get_site(
   local_user_view: Option<LocalUserView>,
   context: Data<LemmyContext>,
 ) -> Result<Json<GetSiteResponse>, LemmyError> {
-  let site_view = SiteView::read_local(&mut context.pool()).await?;
+  static CACHE: Lazy<Cache<(), GetSiteResponse>> = Lazy::new(|| {
+    Cache::builder()
+      .max_capacity(1)
+      .time_to_live(Duration::from_secs(60))
+      .build()
+  });
 
-  let admins = PersonView::admins(&mut context.pool()).await?;
+  // This data is independent from the user account so we can cache it across requests
+  let mut site_response = CACHE
```

Comment:
This one scares me a bit, especially with things like site updates, tagline updates, admin updates, custom emoji updates, etc, not being immediately reflected.

Maybe we should push the caching stuff in here back, but leave the concurrency.

Reply:
There is no reason to read this data over and over again from the database considering that its not personalized and this endpoint gets called a lot. Ive changed the ttl to one second now, that wont be noticable and will still give a major benefit.

Reply:
Mmmk.